src/app/utils/MGPMap.ts

```diff
@@ -3,17 +3,31 @@ import { Comparable, comparableEquals } from './Comparable';
 import { MGPSet } from './MGPSet';
 import { assert } from './utils';
 
-export class MGPMap<K extends Comparable, V extends Comparable> {
+export class MGPMap<K extends NonNullable<Comparable>, V extends NonNullable<unknown>> {
+    public static groupByValue<K extends NonNullable<Comparable>, V extends NonNullable<Comparable>>(map: MGPMap<K, V>)
```

Comment:
Why the NonNullable, and why the patter MyClass.method(myInstance) ?

Reply:
refactorisé avec ReversibleMap